var/ramble/repos/builtin/applications/intel-mpi-benchmarks/application.py

```diff
@@ -41,41 +41,46 @@ class IntelMpiBenchmarks(SpackApplication):
                '-genv I_MPI_FABRICS=shm:tcp '
                '-genv I_MPI_PIN_PROCESSOR_LIST=0 '
                'IMB-MPI1 {pingpong_type} -msglog {msglog_min}:{msglog_max} '
-               '-iter {num_iterations} -show_tail on -dumpfile {output_file}',
+               '-iter {num_iterations} {additional_args}',
                use_mpi=True)
 
-    # FIXME: how best to deal with the desirded behavior of <2*num_cores> -ppn <num_cores>
     executable('multi-pingpong',
                '-genv I_MPI_FABRICS=shm:tcp '
                'IMB-MPI1 Pingpong -msglog {msglog_min}:{msglog_max} '
-               '-iter {num_iteration} -multi 0 -map 2*{num_cores} -show_tail on',
+               '-iter {num_iterations} -multi {multi_val} -map {map_args} {additional_args}',
                use_mpi=True)
 
+    workload_variable('num_cores',
```

Comment:
I don't see any references to `num_cores` in the `multi-pingpong` executable definition on line 47.  Where would this variable get used?

Reply:
There is a fairly subtle layer of indirection here. variables can refer to each other. This is consumed here:

```
 63     workload_variable('map_args',
 64               default='{num_cores}x2',
```

Reply:
Got it, thanks!